docs/chapter-12.rst

```diff
@@ -2,80 +2,1457 @@
+        csrf_session=None,
+        csrf_protection=True,
+        lifespan=None,
+        signing_info=None,
+        ):
+
+Where:
+
+-  ``table``: a DAL table or a list of fields
+-  ``record``: a DAL record or record id
+-  ``readonly``: set to True to make a readonly form
+-  ``deletable``: set to False to disallow deletion of record
+-  ``formstyle``: a function that renders the form using helpers. Can be FormStyleDefault (default),
+   FormStyleBulma or FormStyleBootstrap4 
+-  ``dbio``: set to False to prevent any DB writes
+-  ``keep_values``: if set to true, it remembers the values of the previously submitted form
+-  ``form_name``: the optional name of this form
+-  ``hidden``: a dictionary of hidden fields that is added to the form
+-  ``validation``: an optional validator, see :ref:`Validation functions`
+-  ``csrf_session``: if None, no csrf token is added.  If a session, then a CSRF token is added and verified
+-  ``lifespan``: lifespan of CSRF token in seconds, to limit form validity
+-  ``signing_info``: information that should not change between when the CSRF token is signed and verified
+  
+
+A minimal form example
+----------------------
+
+Let's start with a minimal working form example. Create a new minimal app called ``form_minimal`` :
+
+.. code:: python
+
+    # in form_minimal/__init__.py
+    from py4web import action, Field, redirect, URL
+    from py4web.utils.form import Form
+    from pydal.validators import IS_NOT_EMPTY
+
+
+    @action('index', method=['GET', 'POST'])
+    @action.uses('form_minimal.html')
+    def index():
+        form = Form([
+            Field('product_name'),
+            Field('product_quantity', 'integer', requires=IS_NOT_EMPTY()),
+            ])
+        if form.accepted:
+            # Do something with form.vars['product_name'] and form.vars['product_quantity']
+            redirect(URL('accepted'))
+        if form.errors:
+            # display message error
+            redirect(URL('not_accepted'))
+        return dict(form=form)
+
+    @action("accepted")
+    def accepted():
+        return "form_example accepted"
+
+
+    @action("not_accepted")
+    def not_accepted():
+        return "form_example NOT accepted"
+
+
+Also, you need to create a file inside the app called ``templates/form_minimal.html``
+that just contains the line:
+
+.. code:: html
+
+    [[=form]]
+
+
+Then reload py4web and visit http://127.0.0.1:8000/form_minimal - you'll get the Form page:
+
+.. image:: images/form1.png
+ 
+Note that:
+
+
+-  ``Form`` is a class contained in the ``py4web.utils.form`` module
+-  it's possible to use **form validators** like ``IS_NOT_EMPTY``, see :ref:`Form validation` later.
+   They are imported from the ``pydal.validators`` module
+-  it's normally important to use both the **GET** and the **POST** methods in the action where
+   the form is contained
+
+
+This example it's not so useful because it's not using a database, a template or the session management.
```

Comment:
"This example **is** not so useful" sounds better than "This example it's not so useful."